### ISSUE TITLE
Restoring Microsoft.NETCore.App 1.1.0--preview1-001069-00 produces downgrade warnings

### ISSUE THREAD

**Issue by pranavkm** [dotnet/corefx#12114]
* Multiple versions of the same package?
```
warn : Detected package downgrade: runtime.native.System.Security.Cryptography.OpenSsl from 4.1.1-preview1-24527-03 to 4.0.1-preview1-24527-03
warn :  restore-projects (>= 1.0.0) -> Microsoft.NETCore.App (>= 1.1.0-preview1-001069-00) -> System.Net.Security (>= 4.3.0-preview1-24527-03) -> System.Runtime.Extensions (>= 4.1.0) -> runtime.unix.System.Runtime.Extensions (>= 4.3.0-preview1-24527-03) -> runtime.native.System.Security.Cryptography.OpenSsl (>= 4.1.1-preview1-24527-03)
warn :  restore-projects (>= 1.0.0) -> Microsoft.NETCore.App (>= 1.1.0-preview1-001069-00) -> System.Net.Security (>= 4.3.0-preview1-24527-03) -> runtime.native.System.Security.Cryptography.OpenSsl (>= 4.0.1-preview1-24527-03) 
```

* Incorrect version
```
warn : Detected package downgrade: Microsoft.NETCore.DotNetHostResolver from 1.2.0-beta-000984-00 to 1.1.0-preview1-001069-00
warn :  restore-projects (>= 1.0.0) -> Microsoft.NETCore.App (>= 1.1.0-preview1-001069-00) -> Microsoft.NETCore.DotNetHostPolicy (>= 1.1.0-preview1-001069-00) -> runtime.win7-x64.Microsoft.NETCore.DotNetHostPolicy (>= 1.1.0-preview1-001069-00) -> Microsoft.NETCore.DotNetHostResolver (>= 1.2.0-beta-000984-00)
warn :  restore-projects (>= 1.0.0) -> Microsoft.NETCore.App (>= 1.1.0-preview1-001069-00) -> Microsoft.NETCore.DotNetHostPolicy (>= 1.1.0-preview1-001069-00) -> Microsoft.NETCore.DotNetHostResolver (>= 1.1.0-preview1-001069-00)

warn : Detected package downgrade: Microsoft.NETCore.DotNetHost from 1.2.0-beta-000984-00 to 1.1.0-preview1-001069-00
warn :  restore-projects (>= 1.0.0) -> Microsoft.NETCore.App (>= 1.1.0-preview1-001069-00) -> Microsoft.NETCore.DotNetHostPolicy (>= 1.1.0-preview1-001069-00) -> Microsoft.NETCore.DotNetHostResolver (>= 1.1.0-preview1-001069-00) -> runtime.win7-x64.Microsoft.NETCore.DotNetHostResolver (>= 1.1.0-preview1-001069-00) -> Microsoft.NETCore.DotNetHost (>= 1.2.0-beta-000984-00)
warn :  restore-projects (>= 1.0.0) -> Microsoft.NETCore.App (>= 1.1.0-preview1-001069-00) -> Microsoft.NETCore.DotNetHostPolicy (>= 1.1.0-preview1-001069-00) -> Microsoft.NETCore.DotNetHostResolver (>= 1.1.0-preview1-001069-00) -> Microsoft.NETCore.DotNetHost (>= 1.1.0-preview1-001069-00)

```

**Comment by ericstj** [comment]
For runtime.native.System.Security.Cryptography.OpenSsl it was missing an entry in our packageIndex for the baseline version.

For the host that's something Senthil will have to look at.  I'll open an issue in the core-setup repo.

**Comment by pranavkm** [comment]
I see this warning for lots of packages. Here's two of them

```
warn : Detected package downgrade: System.Runtime from 4.3.0-preview1-24527-03 to 4.1.0 
warn : restore-projects (>= 1.0.0) -> System.Net.Http.WinHttpHandler (>= 4.3.0-preview1-24527-03) -> System.Collections (>= 4.0.11) -> runtime.any.System.Collections (>= 4.3.0-preview1-24527-03) -> System.Runtime (>= 4.3.0-preview1-24527-03) 
warn : restore-projects (>= 1.0.0) -> System.Net.Http.WinHttpHandler (>= 4.3.0-preview1-24527-03) -> System.Runtime (>= 4.1.0)
warn : Detected package downgrade: System.Runtime from 4.3.0-preview1-24527-03 to 4.1.0 
warn : restore-projects (>= 1.0.0) -> System.Net.WebSockets.Client (>= 4.3.0-preview1-24527-03) -> System.Collections (>= 4.0.11) -> runtime.any.System.Collections (>= 4.3.0-preview1-24527-03) -> System.Runtime (>= 4.3.0-preview1-24527-03) 
warn : restore-projects (>= 1.0.0) -> System.Net.WebSockets.Client (>= 4.3.0-preview1-24527-03) -> System.Runtime (>= 4.1.0)
```

**Comment by ericstj** [comment]
Hmm, those seem like NuGet being too aggressive with the downgrade warnings.  The higher version came from a lineup.  This will impact public packages and ASP.NET packages as well.

Consider a package built against RTM referencing S.R & S.C at 1.0.  We update the lineup in 1.1.  Now runtime.S.C depends on S.R 1.1 and NuGet flags that as a downgrade.  /cc @rrelyea @davidfowl

**Comment by ericstj** [comment]
Fixed with #12118

**Comment by RussKie** [comment]
I'm having similar issues https://github.com/dotnet/corefx/issues/14110
